packages/react-ui/components/TokenInput/TokenInput.tsx

```diff
@@ -757,9 +804,24 @@ export class TokenInput<T = string> extends React.PureComponent<TokenInputProps<
     }
   };
 
+  private checkAutocompleteForCorrectItem = () => {
+    const { inputValue, autocompleteItems } = this.state;
+    const { valueToString } = this.props;
+    if (autocompleteItems && autocompleteItems.length > 0) {
+      return (
+        autocompleteItems.length === 1 &&
+        autocompleteItems.find(item => {
+          return valueToString(item) === inputValue;
+        })
+      );
+    }
```

Comment:
По-моему, здесь можно заметно упростить логику.
```suggestion
    if (autocompleteItems && autocompleteItems.length === 1) {
      return autocompleteItems.find(item => valueToString(item) === inputValue);
    }
```

Reply:
отрефакторено